src/api/app/controllers/webui/cloud/upload_jobs_controller.rb

```diff
@@ -0,0 +1,72 @@
+module Webui
+  module Cloud
+    class UploadJobsController < WebuiController
+      before_action :require_login
+      before_action -> { feature_active?(:cloud_upload) }
+      before_action :set_user
+      before_action :validate_configuration_presence
+      before_action :set_breadcrump
+
+      def index
+        @upload_jobs = @user.upload_jobs
+      end
+
+      def new
+        @project_name = params[:project]
+        @package_name = params[:package]
+        @repository_name = params[:repository]
+        @architecture_name = params[:architecture]
+        @file_name = params[:filename]
+      end
+
+      def create
+        begin
+          result = Backend::Api::Cloud.upload(
+            permitted_params[:project],
+            permitted_params[:package],
+            permitted_params[:repository],
+            permitted_params[:architecture],
+            permitted_params[:filename],
+            @user,
+            permitted_params[:region]
+          )
+        rescue ActiveXML::Transport::Error, Timeout::Error => exception
+          flash[:error] = "Failed to create upload job: #{exception.message}"
+          redirect_back(fallback_location: root_path) && return
+        end
+
+        result = Xmlhash.parse(result)
+        @upload_job = @user.upload_jobs.build(
+          job_id: result['name'],
+          user: @user
+        )
```

Comment:
I would move the code from line 23 to here into a model or library as this seems to be part of the upload part.

Reply:
Moved now into UploadJob and created Backend::UploadJob and User::UploadJob as discussed with @evanrolfe